gossipd/routing.c

```diff
@@ -344,13 +344,33 @@ static void clear_bfg(struct node_map *nodes)
 	}
 }
 
-static u64 connection_fee(const struct half_chan *c, u64 msatoshi)
+/**
+ * @brief connection_fee
+ *        Compute the fee for the connection.
+ *
+ * @param from - the node that is the ultimate source of the
+ *        payment.
+ * @param chan - the channel to consider.
+ * @param idx - which half of the channel to consider. 0 or 1.
```

Comment:
We commonly call this direction.

Reply:
The data structure is called `half_chan` and the field is `half[2]`, but I suppose `direction (half_chan)` would be okay.